src/pretix/plugins/sendmail/tasks.py

```diff
@@ -36,9 +36,8 @@ def send_mails(event: Event, user: int, subject: dict, message: dict, orders: li
 
                 if filter_checkins:
                     checkins = list(p.checkins.all())
-                    allowed = (
-                        (not_checked_in and not checkins)
-                        or (any(c.list_id in checkin_lists for c in checkins))
+                    allowed = not checkins if not_checked_in else (
+                        any(c.list_id in checkin_lists for c in checkins)
```

Comment:
The failing tests seem to be related to this one. Note that previously ``allowed`` was ``True`` when ``not_checked_in = True, checkins = True, any(…) = True``) and now it no longer is.